sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -79,36 +107,28 @@ public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
     public static EmbeddingItem fromJson(JsonReader jsonReader) throws IOException {
         return jsonReader.readObject(reader -> {
             String embedding = null;
+            List<Double> embeddingInDouble = null;
             int promptIndex = 0;
             while (reader.nextToken() != JsonToken.END_OBJECT) {
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
                 if ("embedding".equals(fieldName)) {
-                    embedding = reader.getString();
+                    JsonToken jsonToken = reader.currentToken();
+                    if (jsonToken == JsonToken.STRING) {
+                        embedding = reader.getString();
+                    } else if (jsonToken == JsonToken.START_ARRAY) {
+                        embeddingInDouble = reader.readArray(JsonReader::getDouble);
```

Comment:
Can we convert this to base64 string here instead of having the private constructor?

Reply:
We have to keep on private constructor to skip the TypeSpec validaton. From this snapshot you can see actually List<Double> is what generated but not String. So Keep editing the private constructor and keep generated annotation for List<DouBle> 
![image](https://github.com/user-attachments/assets/86abdcd6-57ce-488b-98d9-47a4e30256fe)

I have added a helper method to convert the List<Float> to base64 encoded string. So when user pass a list of floats, they will have base64encoded string as well.